python/python/lance/ray/distribute_task.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+from typing import Any, Callable, Dict, List, Optional
+
+import lance
+
+# ==============================================================================
+# Message Structure Constants
+# ==============================================================================
+TASK_ID_KEY = "task_id"
+PARTITION_KEY = "partition"
+
+# ==============================================================================
+# Data Component Keys
+# ==============================================================================
+FRAGMENT_KEY = "fragment"
+SCHEMA_KEY = "schema"
+
+# ==============================================================================
+# Operation Parameters
+# ==============================================================================
+PARAMS_KEY = "params"
+ACTION_KEY = "action"
+READ_COLUMNS_KEY = "read_columns"
+
+# ==============================================================================
+# Execution Metadata
+# ==============================================================================
+OPERATION_TYPE_KEY = "operation_type"
+VERSION_KEY = "version"
+
+
+class TaskInput:
+    """Container for task execution parameters and metadata."""
+
+    def __init__(
+        self,
+        task_id: str,
+        fn: Callable,
+        fragment: Any,
+        params: Optional[Dict[str, Any]] = None,
+    ):
+        self.task_id = task_id
+        self.fn = fn
+        self.fragment = fragment
+        self.params = params or {}
+
+
+class FragmentTask:
+    """Base class for distributed data processing tasks."""
+
+    def __init__(self, task_input: TaskInput):
+        self._fn = task_input.fn
+        self._task_id = task_input.task_id
+        self._fragment = task_input.fragment
+        self._params = task_input.params
```

Comment:
Should we just do this?

```
self._task_input = task_input
```

This way we don't have to duplicate most properties.